### PR TITLE
Allowed users to select more than two raid mods in the mod picker

### LINKED ISSUE #6402: Unable to select more than 2 raid mods at a time in the ModPicker sheet
**What version of DIM are you using?**
Repros on live and beta

**What Browser and OS are you using?**
Repros on Chrome 87 & 88

**Describe the bug and how to make it happen**
After selecting raid mods, the ModPicker sheet incorrectly limits raid mods to only 2 at a time regardless of the raid.

_Repro steps:_
1) In game, equip 5 pieces of raid gear from the garden of salvation raid.
1) In game, apply the Relay Defender mod to all 5 to prove it's legal to have 5 of that mod in a loadout.
1) In DIM, open Loadout Manager, then tap Add Mod.
1) Attempt to add Relay Defender five times.

_Expected Results:_
I can select up to 5 relay defenders.

_Actual Results:_
I can select only 2.

### PR DESCRIPTION
**Pull request by ryan-rushton**
Fixes #6402

### CODE REVIEW

**Review by nev-r — APPROVED** [review]
perfect thank you

File: src/app/loadout-builder/filter/ModPicker.tsx [hunk 1]
```diff
@@ -50,6 +50,14 @@ const sortMods = chainComparator<LockedArmor2Mod>(
   compareBy((l) => l.modDef.displayProperties.name)
 );
 
+function isGeneralOrOtherOrRaid(category: ModPickerCategory) {
```
Comment (bhollis) [inline src/app/loadout-builder/filter/ModPicker.tsx:53]: Unfortunately this function name tells me what it *does* but not what it's *for*. What are we trying to represent with this, and why does it imply you can select 5 mods vs 2?

Comment (ryan-rushton) [reply src/app/loadout-builder/filter/ModPicker.tsx:53]: Yeah I know exactly what you mean and it bothered me as well. Problem is it serves two complete seperate purposes. Whether we allow 2 or 5 total mods to be selected OR whether we need to care about mod energies matching. I would just use the condition directly but its pretty large to put into a ternary imo. Let me know what you want to do with it and happy to go that route. I'm not happy with most things I have come up with.

Comment (chrisdouglass) [reply src/app/loadout-builder/filter/ModPicker.tsx:53]: In order to unblock this PR and address the comment, for now can this be renamed to "isSlotSpecificCategory" or something that states its purpose?

**Comment by ryan-rushton** [comment]
@bhollis, going to merge this and the one @chrisdouglass did. I will start some cleanup PR's straight after.

**Pull request merged**